### DISCUSSION TITLE
Headless UI: Unstyled, Accessible UI Components by Tailwind Labs

### DISCUSSION THREAD

**Discussion by adamwathan** [tailwindlabs/tailwindcss#2508] [Show and tell]
I just published a blog post announcing our latest open-source project [Headless UI](https://headlessui.dev).

[**Read "Headless UI: Unstyled, Accessible UI Components"**](https://blog.tailwindcss.com/headless-ui-unstyled-accessible-ui-components)

It's a set of completely unstyled, fully accessible UI components for React, Vue, and soon Alpine.js that make is super easy to build custom components without sacrificing the ability to style them with Tailwind.

Check it out and let us know what you think!

**Comment by renatodeleao** [comment]
Your 2017 post and the following course `advanced-vue-component-design` are still to this day the most inspirational resources on building reusable components for me! (Don't tell anyone but sometimes I go back the the course videos to remember a few tricks)

I can definitely relate with this project, and after seeing some promising `vue` projects in this area fall behind, it's exciting to see a team like `tailwind` behind one!  I'll follow it closely! Keep it up :shipit:

**Comment by jackguoAtJogg** [comment]
I wonder if I can only style the components with tailwind or I can do it with css-in-js library

**Reply by adamwathan** [reply]
You can style them with anything you want! This library is totally detached from Tailwind 👍🏻

**Reply by jackguoAtJogg** [reply]
Thats dope, I'll give a try very soon! Thank you and I really like tailwind as well.

**Comment by JoseJuan81** [comment]
Does anybody can collaborates with this project or is a closed one?
I have some vue components made them with this concept.

**Reply by eric-hc** [reply]
It's [open source](https://github.com/tailwindlabs/headlessui)!

**Reply by JoseJuan81** [reply]
@eric-hc please, can you give a orientation to collaborate?, what is the standard or rules to collaborate?

**Reply by eric-hc** [reply]
@RobinMalfait would likely be the person to ask

**Reply by simonswiss** [reply]
Hi @JoseJuan81 👋

Thanks for reaching out! I can't speak for Robin or Adam - who are the key contributors to this library - but I would say that at this point, collaboration/contribution isn't _too_ likely to happen. We're still figuring many of the APIs and ergonomics to shape up this library, and it's probably a bit early to start taking external contributions.

Feedback and suggestions are most definitely appreciated, but I don't think this library specifically is the best place for contributing full components, at least right now.

I might be wrong though, and suggestions/ideas are never frowned upon! In any case, please start with opening an issue and starting a discussion, rather than submitting a PR 👍

**Comment by miller-productions** [comment]
Looks like something excellent on the way! Great idea @adamwathan

Just wondering, what is the relationship between HeadlessUI and TailwindUI?

In other words, when would someone use one over the other?

e.g.

https://tailwindui.com/components/application-ui/forms/select-menus

vs

https://github.com/tailwindlabs/headlessui/tree/develop/packages/%40headlessui-react#listbox-select

**Reply by RobinMalfait** [reply]
You don't choose one over the other, you can use them together.

Headless UI, is headless, no styles at all, just functionality. Tailwind UI is html + classes based on Tailwind CSS. 
If you want to create a good looking Listbox, then you can use both together. 

In your second link, if you look at the example CodeSandbox, then you will notice that we used Tailwind UI classes on top of Headless UI functionality.

**Reply by miller-productions** [reply]
Thanks Robin,

That's excellent. Exactly how I started using it.

**Comment by sethhorsley** [comment]
> I just published a blog post announcing our latest open-source project [Headless UI](https://headlessui.dev).
> 
> [**Read "Headless UI: Unstyled, Accessible UI Components"**](https://blog.tailwindcss.com/headless-ui-unstyled-accessible-ui-components)
> 
> It's a set of completely unstyled, fully accessible UI components for React, Vue, and soon Alpine.js that make is super easy to build custom components without sacrificing the ability to style them with Tailwind.
> 
> Check it out and let us know what you think!

Hey @adamwathan i was wondering if there has been any progress on the alpine.js version?

**Reply by hugronaphor** [reply]
Most of components have examples at https://alpinejs.dev/components
